stake-pool/cli/src/main.rs

```diff
@@ -2018,17 +2472,33 @@ fn main() {
                 denominator,
                 numerator,
             };
-            command_set_fee(&config, &stake_pool_address, new_fee)
+            match arg_matches.value_of("fee_type").unwrap() {
```

Comment:
This change-set is already getting pretty massive, but we could reduce the number of instructions and simplify the code quite a bit if we actually add this as an enum, `FeeType`.  That can be passed into generic `SetFee` and `SetReferralFee`, which would reduce the number of instructions from 5 to 2.  It would also get rid of that `for_stake_deposit` bool, which isn't very clear unfortunately.  What do you think?

Reply:
What do you think of combining everything into 1 enum?

```
enum FeeType {
  ReferreralSol(u8),
  ReferralStake(u8),
  Epoch(Fee),
  Withdrawal(Fee),
  Deposit(Fee),
}
```

Reply:
Yeah sure, that works!